src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,192 @@
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndexFits(index);
+        Node<T> current = first;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                return current.item;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
+        checkIndexFits(index);
+        Node<T> current = first;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                T result = current.item;
+                current.item = value;
+                return result;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexFits(index);
+        Node<T> toRemove;
+        if (index == 0) {
+            toRemove = first;
+            first = first.next;
+            if (first != null) {
+                first.prev = null;
+            } else {
+                last = null;
+            }
+        } else if (index == size - 1) {
+            toRemove = last;
+            last = last.prev;
+            if (last != null) {
+                last.next = null;
+            } else {
+                first = null;
+            }
+        } else {
+            Node<T> current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            toRemove = current;
+            Node<T> prevNode = current.prev;
+            Node<T> nextNode = current.next;
+
+            prevNode.next = nextNode;
+            nextNode.prev = prevNode;
+        }
+
+        size--;
+        return toRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        if (first == null) {
+            return false;
+        }
+        Node<T> current = first;
+        while (current != null) {
+            if ((object == null && current.item == null)
+                    || (object != null && object.equals(current.item))) {
+                if (current.prev != null) {
+                    current.prev.next = current.next;
+                } else {
+                    first = current.next;
+                }
+
+                if (current.next != null) {
+                    current.next.prev = current.prev;
+                } else {
+                    last = current.prev;
+                }
+                size--;
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
```

Comment:
The 'checkIndex' method name could be more informative, such as 'ensureIndexWithinBounds'.